src/main.rs

```diff
@@ -151,15 +177,46 @@ fn main() -> ! {
     let echo_socket_handle = socket_set.add(socket);
     let coap_socket_handle = socket_set.add(coap_socket);
 
+    let have_certchain = have_certchain(&mut trussed_fobnail_client);
     let have_rims = have_rims(&mut trussed_fobnail_client);
     let mut operation_mode = OperationMode::new(&mut trussed_fobnail_client);
-    operation_mode = if !have_rims {
+
+    operation_mode = if !have_certchain {
+        operation_mode.token_provisioning()
+    } else if !have_rims {
         operation_mode.provisioning()
     } else {
         operation_mode.attestation()
     };
 
+    let mut last_button_state = false;
+    let mut button_press_time = 0;
     loop {
+        let now = pal::timer::get_time_ms() as u64;
+        // Pressing button for 10 s triggers Fobnail Token provisioning.
+        if now - button_press_time > 10000
+            && !matches!(operation_mode, OperationMode::TokenProvisioning(_))
```

Comment:
This only checks if button state changed more than 10s ago, but doesn't check its current state. It probably will purge the device 10s after button is released, even when it was pressed for shorter period.